core/chains/evm/client/node_fsm.go

```diff
@@ -97,6 +123,7 @@ func (n *node) declareAlive() {
 }
 
 func (n *node) transitionToAlive(fn func()) {
+	promEVMPoolRPCNodeTransitionsToAlive.WithLabelValues(n.chainID.String(), n.name).Inc()
```

Comment:
Maybe insignificant, but should this happen afterwards and only if the transition actually happened? i.e. not the short circuit return or panic paths?

Reply:
Good point, I was trying to avoid putting any additional logic inside the lock.

I think its ok since these metrics are statistical in nature and it ought to be very rare that it would report whilst hitting a different path